caffe2/serialize/inline_container.h

```diff
@@ -119,135 +134,202 @@ class PyTorchStreamReader {
+    return true;
+  }
+
+  bool readAndValidateFileFooter() {
     // Seek to location of file footer. We've already validated that the file
     // length is a multiple of the alignment size
-    cursor = file_size - kFieldAlignment;
-    in.seekg(cursor);
+    cursor_ = file_size_ - kFieldAlignment;
+    in_.seekg(cursor_);
     auto tag = read64BitIntegerLittleEndian();
     if (tag != RecordTags::FOOTER) {
       throw std::runtime_error("File footer has wrong record type. Is this"
                                " file corrupted?");
     }
-    last_record_offset = read64BitIntegerLittleEndian();
-    if (last_record_offset > file_size) {
+    last_record_offset_ = read64BitIntegerLittleEndian();
+    if (last_record_offset_ > file_size_) {
       throw std::runtime_error("Offset of last record is higher than the size"
                                " of the file! Is this file corrupted?");
     }
+    return true;
   }
 };
 
-class PyTorchStreamWriter {
+class PyTorchStreamWriter final {
  public:
-  PyTorchStreamWriter(std::ostream& out_) : out(out_) {
+  PyTorchStreamWriter(std::ostream& out) : out_(out) {
     writeFileHeader();
     // In the case that we do not write any records into this file, the last
     // record index written into the footer will point to the footer itself.
-    last_record_idx = cursor;
+    last_record_idx_ = cursor_;
   }
+
   uint64_t writeRecord(const char* data, size_t size) {
-    JIT_ASSERT(!finalized);
-    uint64_t record_offset = cursor;
-    last_record_idx = record_offset;
+    CAFFE_ENFORCE(!finalized_, "should not be finalized!");
+    uint64_t record_offset = cursor_;
+    last_record_idx_ = record_offset;
     write64BitIntegerLittleEndian(RecordTags::STORAGE);
     write64BitIntegerLittleEndian(size);
     padToNextAlignmentBoundary();
     writeBuffer(data, size);
     padToNextAlignmentBoundary();
     return record_offset;
   }
+
   void writeEndOfFile() {
-    JIT_ASSERT(!finalized);
+    CAFFE_ENFORCE(!finalized_, "cannot finalize again!");
     writeFileFooter();
-    finalized = true;
+    finalized_ = true;
+  }
+
+  int64_t getCurrentSize() {
+    return static_cast<int64_t>(cursor_);
+  }
+
+  bool finalized() {
+    return finalized_;
   }
+
   ~PyTorchStreamWriter() {
-    if (!finalized) {
+    if (!finalized_) {
       writeEndOfFile();
     }
   }
+
  private:
-  std::ostream& out;
-  size_t cursor = 0;
-  bool finalized = false;
-  size_t last_record_idx = 0;
+  std::ostream& out_;
+  size_t cursor_ = 0;
+  bool finalized_ = false;
+  size_t last_record_idx_ = 0;
 
   // Utility functions
   void write64BitIntegerLittleEndian(const uint64_t value) {
     // TODO endian swap on platforms that need it?
-    out.write(reinterpret_cast<const char *>(&value), 8);
-    cursor += 8u;
+    out_.write(reinterpret_cast<const char*>(&value), 8);
+    cursor_ += 8u;
   }
 
   void writePad(const size_t num_bytes) {
     static std::vector<char> pad_buffer(/*count=*/kFieldAlignment,
```

Comment:
I'm not sure why using static std::vector here. Consider make it a member of this class?